lib/Dialect/FIRRTL/FIRRTLAnnotations.cpp

```diff
@@ -362,19 +362,20 @@ bool AnnotationSet::removeAnnotations(Operation *op, StringRef className) {
       op, [&](Annotation a) { return (a.getClass() == className); });
 }
 
-/// Remove all port annotations from a module for which `predicate` returns
-/// true.
+/// Remove all port annotations from a module or extmodule for which `predicate`
+/// returns true.
 bool AnnotationSet::removePortAnnotations(
-    FModuleOp module,
+    Operation *module,
     llvm::function_ref<bool(unsigned, Annotation)> predicate) {
   // We need to reserve some space to gather the remaining attributes, without
   // the removed ones.
   SmallVector<DictionaryAttr, 8> filteredArgAttrs;
-  filteredArgAttrs.reserve(module.getNumArguments());
+  FunctionType moduleF = mlir::function_like_impl::getFunctionType(module);
+  filteredArgAttrs.reserve(moduleF.getNumInputs());
```

Comment:
Very nice! I kept asking myself how to do this properly :see_no_evil:

Reply:
I get hung up on the seemingly weird ergonomics of being given an `Operation *` and wanting to work with it like the sum of `FModule` and `FExtModule`.  `TypeSwitch` kind of fakes this, but wasn't appropriate here.  I then keep finding new methods that seem to help...